src/v/debug_bundle/types.cc

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright 2024 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#include "types.h"
+
+#include <seastar/util/variant_utils.hh>
+
+auto fmt::formatter<debug_bundle::special_date>::format(
+  debug_bundle::special_date d, format_context& ctx) const
+  -> format_context::iterator {
+    return formatter<string_view>::format(debug_bundle::to_string_view(d), ctx);
+}
+
+fmt::format_context::iterator
+fmt::formatter<debug_bundle::time_variant>::format(
+  const debug_bundle::time_variant& t, format_context& ctx) const {
+    return ss::visit(
+      t,
+      [&ctx](const debug_bundle::debug_bundle_timepoint& t) {
+          auto tt = debug_bundle::debug_bundle_clock::to_time_t(t);
+          std::stringstream ss;
+          ss << std::put_time(std::localtime(&tt), "%FT%T");
+          return fmt::format_to(ctx.out(), "{}", ss.str());
```

Comment:
Ugh, timezones... what I would like is for the user to provide timestamps relative to the same timezone as their cluster (so they don't have to convert to UTC) so that same time stamp can be provided to rpk via command line arguments.  This timestamp is used by journal to only return logs in a specified time range.